metadata-ingestion/src/datahub/ingestion/sink/datahub_rest.py

```diff
@@ -195,13 +197,23 @@ def write_record_async(
         write_callback: WriteCallback,
     ) -> None:
         record = record_envelope.record
-        write_future = self.executor.submit(self.emitter.emit, record)
-        write_future.add_done_callback(
-            functools.partial(
-                self._write_done_callback, record_envelope, write_callback
+        try:
+            write_future = self.executor.submit(self.emitter.emit, record)
+            write_future.add_done_callback(
+                functools.partial(
+                    self._write_done_callback, record_envelope, write_callback
+                )
             )
-        )
-        self.report.pending_requests += 1
+            self.report.pending_requests += 1
+        except RuntimeError:
```

Comment:
when might this happen?

Also if the message is submitted and then fails afterwards and triggers the callback with a failure, this error handling logic won't get hit